### ISSUE TITLE
Dead code not detected

### ISSUE THREAD

**Issue by 64kramsystem** [rust-lang/rust#101379]
In a project of mine, some dead code is not detected.

It's considerably time consuming to provide a minimal test case, because I'd have to progressively rip out all the functions in the codebase, hoping that, in the meanwhile, the bug still reproduces; for this reason, I'm providing the steps required to reproduce the problem, which is deterministic.

Perform the following steps:

```sh
git clone git@github.com:64kramsystem/catacomb_ii-64k
cd catacomb_ii-64k
git checkout dead_code_not_detected

# No problems detected.
cargo check

# This verifies that the function is dead; no problems detected as well
perl -i -pe 's/_printhexb/_phonyname/' src/pcrlib_c.rs
cargo check
```

The current output is a clean build.

Ideally rustc should:

- detect that there is some dead code; see the [_printhexb function](https://github.com/64kramsystem/catacomb_ii-64k/blob/dead_code_not_detected/src/pcrlib_c.rs#L1014-L1017)
  - there are other functions that are dead (`_printhex`, `_printbin`, maybe others)

The rust version used is:

```
rustc 1.64.0-nightly (fe3342816 2022-08-01)
binary: rustc
commit-hash: fe3342816a282949f014caa05ea2e669ff9d3d3c
commit-date: 2022-08-01
host: x86_64-unknown-linux-gnu
release: 1.64.0-nightly
LLVM version: 14.0.6
```

**Comment by ChrisDenton** [comment]
Prefixing with an underscore silences the dead code lint. This is intended behavior, see https://doc.rust-lang.org/rustc/lints/listing/warn-by-default.html#dead-code

**Comment by 64kramsystem** [comment]
Uh! I didn't know that. Sorry :grimacing:
